host_core/lib/host_core/actors/actor_module.ex

```diff
@@ -280,22 +294,31 @@ defmodule HostCore.Actors.ActorModule do
     {:ok, agent} =
       Agent.start_link(fn -> %State{claims: claims, instance_id: UUID.uuid4(), healthy: false} end)
 
-    prefix = HostCore.Host.lattice_prefix()
-    topic = "wasmbus.rpc.#{prefix}.#{claims.public_key}"
-
-    Logger.debug("Subscribing to #{topic}")
-    {:ok, subscription} = Gnat.sub(:lattice_nats, self(), topic, queue_group: topic)
-    Agent.update(agent, fn state -> %State{state | subscription: subscription, ociref: oci} end)
-
     imports = %{
       wapc: Imports.wapc_imports(agent),
       wasmbus: Imports.wasmbus_imports(agent)
     }
 
-    publish_oci_map(oci, claims.public_key)
+    case Wasmex.start_link(%{bytes: bytes, imports: imports}) |> prepare_module(agent, oci) do
+      {:ok, agent} ->
+        prefix = HostCore.Host.lattice_prefix()
+        topic = "wasmbus.rpc.#{prefix}.#{claims.public_key}"
+
+        Logger.debug("Subscribing to #{topic}")
+        {:ok, subscription} = Gnat.sub(:lattice_nats, self(), topic, queue_group: topic)
+
+        Agent.update(agent, fn state ->
+          %State{state | subscription: subscription, ociref: oci}
+        end)
 
-    Wasmex.start_link(%{bytes: bytes, imports: imports})
-    |> prepare_module(agent, oci)
+        publish_oci_map(oci, claims.public_key)
+        {:ok, agent}
+
+      {:error, e} ->
+        Logger.error("Failed to start actor: #{inspect(e)}")
+        HostCore.ControlInterface.Server.publish_actor_start_failed(claims.public_key, inspect(e))
```

Comment:
I don't love the double publish but I can't find a better place to centralize the "start failed" logic 🤔 perhaps we can revisit this in the case where Wasmex crashes happen for more than failed imports.

Reply:
this fix actually covers all Wasmex failures, not just failed imports.